Bricks/sync/scope_owned.h

```diff
@@ -0,0 +1,388 @@
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    f(operator*());
+  }
+  template <typename V>
+  V ExclusiveUse(std::function<V(T&)> f) {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    return f(operator*());
+  }
+  template <typename V>
+  V ExclusiveUse(std::function<V(const T&)> f) const {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    return f(operator*());
+  }
+
+  // Exclusive accessors ignoring the `destructing_` flag.
+  // NOT THREAD-SAFE. NEVER THROW.
+  void ExclusiveUseDespitePossiblyDestructing(std::function<void(T&)> f) {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    f(p_actual_instance_->instance_);
+  }
+  void ExclusiveUseDespitePossiblyDestructing(std::function<void(const T&)> f) const {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    f(p_actual_instance_->instance_);
+  }
+  template <typename V>
+  V ExclusiveUseDespitePossiblyDestructing(std::function<V(T&)> f) {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    return f(p_actual_instance_->instance_);
+  }
+  template <typename V>
+  V ExclusiveUseDespitePossiblyDestructing(std::function<V(const T&)> f) const {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    return f(p_actual_instance_->instance_);
+  }
+
+  // Return the number of registered follower users besides the master one.
+  // For unit-testing purposes mostly, but may end up useful. -- D.K.
+  // THREAD-SAFE. NEVER THROWS.
+  size_t NumberOfActiveFollowers() const {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    return p_actual_instance_->followers_.size();
+  }
+
+  // Return the total number of registered follower users registered, with some possibly already out of scope.
+  // For unit-testing purposes mostly, but may end up useful. -- D.K.
+  // THREAD-SAFE. NEVER THROWS.
+  size_t TotalFollowersSpawnedThroughoutLifetime() const {
+    std::lock_guard<std::mutex> lock(p_actual_instance_->mutex_);
+    return p_actual_instance_->total_followers_spawned_throughout_lifetime_;
+  }
+
+ private:
+  void CheckPrivilege() const {
+    if (p_actual_instance_->destructing_) {
+      CURRENT_THROW(InDestructingModeOfTException<T>());
+    }
+  }
+
+  T_INSTANCE* p_actual_instance_;
+  size_t key_;  // The index of the slave user to mark as left the scope, or `0u` if N/A (master or moved away).
+};
+
+namespace impl {
+
+// `ActualInstanceContainer<T>` must be a dedicated struct for `ScopeOwnedByMe<T>` to construct
+// the actual object before initializing the `ScopeOwned<T>` associated with it.
+// `ActualInstanceContainer<T>` is constructible, movable, and nothing else.
+template <typename T>
+struct ActualInstanceContainer {
+  std::unique_ptr<ActualInstance<T>> movable_instance_;
+
+  ActualInstanceContainer(ConstructActualContainerViaMoveConstructor, ActualInstanceContainer&& rhs)
```

Comment:
What is the purpose of requiring the `ConstructActualContainerViaMoveConstructor` token here?